lib/libspl/include/assert.h

```diff
@@ -42,6 +42,17 @@ __assert_c99(const char *expr, const char *file, int line, const char *func)
 }
 #endif  /* __assert_c99 */
 
+#ifndef __assert
```

Comment:
I could not build this with latest master as #4422 does not apply for me anymore.
So I have applied this to the latest stable with our other patches and this seems to build correctly.